lib/classroom_api.rb

```diff
@@ -8,190 +11,229 @@ def initialize(access_token)
 
-    request = Net::HTTP::Get.new(url)
-    request["x-ibm-client-id"] = "#{Rails.application.credentials.um_api[:buildings_client_id]}"
-    request["authorization"] = "Bearer #{@access_token}"
-    request["accept"] = 'application/json'
+      request = Net::HTTP::Get.new(url)
+      request["x-ibm-client-id"] = "#{Rails.application.credentials.um_api[:buildings_client_id]}"
+      request["authorization"] = "Bearer #{@access_token}"
+      request["accept"] = 'application/json'
 
-    response = http.request(request)
-    response_json = JSON.parse(response.read_body)
-    if response_json.present?
-      if response_json['errorCode'].present?
+      response = http.request(request)
+      response_json = JSON.parse(response.read_body)
+
+      if response.code == OK_CODE
+        result['success'] = true
+        result['data'] = response_json['Classrooms']
+      elsif response_json['errorCode'].present?
         result['errorcode'] = response_json['errorCode']
         result['error'] = response_json['errorMessage']
-        result['success'] = false
-      else
-        result['success'] = true
-        result['data'] = response_json['Classrooms']['Classroom']
+      else 
+        result['errorcode'] = "Unknown error"
       end
+    rescue StandardError => e
+      result['errorcode'] = "Exception"
+      result['error'] = e.message
     end
     return result
   end
 
   def update_all_classroom_characteristics
-
-    classrooms = Room.where(rmtyp_description: "Classroom").where.not(facility_code_heprod: nil)
-    number_of_api_calls_per_minutes = 0
-    redo_loop_number = 1
-    classrooms.each do |room|
-      if number_of_api_calls_per_minutes < 400
-        number_of_api_calls_per_minutes += 1
-      else
-        @log.api_logger.debug "update_all_classroom_characteristics, the script sleeps after #{number_of_api_calls_per_minutes} calls"
-        number_of_api_calls_per_minutes = 1
-        sleep(61.seconds)
-      end
-      facility_id = room.facility_code_heprod
-      rmrecnbr = room.rmrecnbr
-      result = get_classroom_characteristics(ERB::Util.url_encode(facility_id))
-      if result['errorcode'] == "ERR429"
-        @log.api_logger.debug "update_all_classroom_characteristics, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls"
-        number_of_api_calls_per_minutes = 0
-        sleep(61.seconds)
-        if redo_loop_number > 9
-          @debug = true
-          @log.api_logger.debug "update_all_classroom_characteristics, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls #{redo_loop_number} times "
-          return @debug
+    begin
+      classrooms = Room.where(rmtyp_description: "Classroom").where.not(facility_code_heprod: nil)
+      number_of_api_calls_per_minutes = 0
+      redo_loop_number = 1
+      classrooms.each do |room|
+        if number_of_api_calls_per_minutes < 400
```

Comment:
Consider creating constant variable for this value.

Reply:
done in [6158ce0](https://github.com/lsa-mis/mi_classrooms/pull/326/commits/6158ce0a5e3b305c2ca803b248e447d8701300f2)